code/game/machinery/kitchen/smartfridge.dm

```diff
@@ -226,17 +168,17 @@
 			overlays += "drying_rack_drying"
 
 /obj/machinery/smartfridge/drying_rack/proc/dry()
-	for(var/datum/data/stored_item/I in item_records)
+	for(var/datum/stored_item/I in item_records)
 		for(var/obj/item/weapon/reagent_containers/food/snacks/S in I.instances)
 			if(S.dry) continue
 			if(S.dried_type == S.type)
 				S.dry = 1
 				S.name = "dried [S.name]"
 				S.color = "#AAAAAA"
-				I.get_product(loc)
+				stock(S)

```

Comment:
S is still part of I.instances in both this and the else case below.
In the first case it means that one can magically dispense the object twice (it'll even teleport across the universe to end up at the machine), in the second it means null entries will end up in the I.instances list.